pkg/local_object_storage/blobstor/fstree/fstree.go

```diff
@@ -503,25 +503,49 @@ func (t *FSTree) GetStream(addr oid.Address) (*objectSDK.Object, io.ReadCloser,
 
 // GetRange implements common.Storage.
 func (t *FSTree) GetRange(addr oid.Address, from uint64, length uint64) ([]byte, error) {
-	obj, err := t.Get(addr)
+	_, reader, err := t.getObjectStream(addr)
 	if err != nil {
 		return nil, err
 	}
+	defer reader.Close()
 
-	payload := obj.Payload()
-	pLen := uint64(len(payload))
-	var to uint64
-	if length != 0 {
-		to = from + length
+	var n int64
+	if seeker, ok := reader.(io.Seeker); ok {
+		n, err = seeker.Seek(int64(from), io.SeekStart)
 	} else {
-		to = pLen
+		n, err = io.CopyN(io.Discard, reader, int64(from))
```

Comment:
I think you can implement proper `Seek()` for `payloadReader`, it'll be faster (position the buffer reader and/or seek in the file) and it can be useful in future as well (think of resetting the stream after transmission).